src/ImageRunModal.jsx

```diff
@@ -32,6 +32,11 @@ const PublishPort = ({ id, item, onChange, idx, removeitem, additem }) =>
     (
         <>
             <InputGroup className='ct-input-group-spacer-sm' id={id}>
+                <TextInput aria-label={_("IP (optional)")}
+                           type='text'
+                           placeholder={_("IP (optional)")}
+                           value={item.IP || ''}
+                           onChange={value => onChange(idx, 'IP', value)} />
```

Comment:
We could also make IP and TCP/UDP is one line and the Port mapping in next, so that they don't squeeze so much. but design improvements come come later on of course.